src/lib/generate-site-name.ts

```diff
@@ -23,22 +23,30 @@ export function generateSiteName( usedSiteNames: string[] ): string {
 		__( 'My Swift Website' ),
 		__( 'My True Website' ),
 	];
-	let proposedName = __( 'My WordPress Website' );
-	let tryCount = 0;
 
-	while ( usedSiteNames.includes( proposedName ) && tryCount < siteNames.length ) {
-		tryCount++;
-		proposedName = siteNames[ Math.floor( Math.random() * siteNames.length ) ];
+	const defaultName = __( 'My WordPress Website' );
+	if ( ! usedSiteNames.includes( defaultName ) ) {
+		return defaultName;
 	}
 
-	return proposedName;
+	const availableNames = siteNames.filter( ( name ) => ! usedSiteNames.includes( name ) );
+	if ( availableNames.length > 0 ) {
+		return availableNames[ Math.floor( Math.random() * availableNames.length ) ];
+	}
+
+	const randomName = siteNames[ Math.floor( Math.random() * siteNames.length ) ];
+	let postfix = 1;
+	while ( usedSiteNames.includes( `${ randomName } ${ postfix }` ) ) {
+		postfix++;
+	}
+	return `${ randomName } ${ postfix }`;
```

Comment:
Should we take into account RTL languages ? For those languages the postfix should be on the left.

Reply:
Would it make sense to always use `My WordPress Website` for subsequent names with the number, so user who uses all random names will keep getting `My WordPress Website 1`, `My WordPress Website 2` and so on?

Reply:
UPDATE, Oh, I guess you meant exactly to use *only* subsequent names of `My WordPress Website` and remove adjective options, right? If so, ignore my previous message and I think that it's good simple approach, so I don't have strong preference.

Reply:
Yes, this is what I meant. So the order would look as follows:
- `My WordPress Website`
- One of `My RANDOM Website` names
- `My WordPress Website 1`
- `My WordPress Website 2`
This seems to be the least confusing to me.